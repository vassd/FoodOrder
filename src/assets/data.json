{
  "restaurant": {
    "name": "3rd rest"
  },
  "menu": {
    "items": [
      {
        "name": "Pizza Margherita",
        "price": 7.00,
        "currency": "RON",
        "ingredients": [
          "tomato sauce",
          "cheese"
        ],
        "category": 0
      },
      {
        "name": "Pizza Prosciutto",
        "price": 10.00,
        "currency": "RON",
        "ingredients": [
          "tomato sauce",
          "cheese",
          "ham"
        ],
        "category": 0
      },
      {
        "name": "Spaghetti Bolognese",
        "price": 9.00,
        "currency": "RON",
        "ingredients": [
        ],
        "category": 1
      },
      {
        "name": "Spaghetti Frutti di Mare",
        "price": 12.00,
        "currency": "RON",
        "ingredients": [
          "with sea food"
        ],
        "category": 1
      }
    ],
    "categories": [
      {
        "id": 0,
        "name": "Pizza"
      },
      {
        "id": 1,
        "name": "Pasta"
      }
    ]
  }
}