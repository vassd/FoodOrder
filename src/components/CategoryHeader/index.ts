export { CategoryHeader } from './CategoryHeader'
export type { Category } from './CategoryHeader'
