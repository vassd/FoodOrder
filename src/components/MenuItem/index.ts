export { MenuItem } from './MenuItem'
export type { MenuItemInterface } from './MenuItem'
