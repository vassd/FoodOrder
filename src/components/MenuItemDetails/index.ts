export { MenuItemDetails } from './MenuItemDetails'
export type { Topping, Crust } from './MenuItemDetails'
