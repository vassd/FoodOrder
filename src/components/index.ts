export { MenuHeader } from './MenuHeader'
export { CategoryHeader } from './CategoryHeader'
export type { Category } from './CategoryHeader'
export { MenuItems } from './MenuItems'
export { MenuItem } from './MenuItem'
export type { MenuItemInterface } from './MenuItem'
