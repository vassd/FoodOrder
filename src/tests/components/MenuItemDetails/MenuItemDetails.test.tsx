import React from 'react'
import '@testing-library/jest-dom'
import { describe, expect, test, vi } from 'vitest'
import { render, screen, fireEvent } from '@testing-library/react'
import { MenuItemDetails, type MenuItemInterface } from 'components'
import data from 'assets/data.json'

describe('MenuItemDetails test', () => {
  const testIsOpen: boolean = true
  const testItem: MenuItemInterface = data.menu.items[0]
  const mockOnClose = vi.fn()

  test('should match snapshot', () => {
    const { container } = render(
      <MenuItemDetails
        isOpen={testIsOpen}
        item={testItem}
        onClose={mockOnClose}
      />
    )

    expect(container).toMatchSnapshot()
  })

  test('should call onClick', () => {
    render(
      <MenuItemDetails
        isOpen={testIsOpen}
        item={testItem}
        onClose={mockOnClose}
      />
    )

    const item = screen.getByLabelText('close dialog')
    fireEvent.click(item)

    expect(mockOnClose).toBeCalled()
  })
})
