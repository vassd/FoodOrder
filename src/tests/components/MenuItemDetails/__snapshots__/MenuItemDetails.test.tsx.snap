// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MenuItemDetails test > should match snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;
